File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.java
import java.util.*;
class Solution {
    public String solution(int n, int t, int m, int p) {
        
        StringBuilder sb = new StringBuilder();
        StringBuilder ans = new StringBuilder();
       
        for(int i = 0; sb.length() <= t*m; i++) {
            sb.append(Integer.toString(i,n));
        }
        for(int i = p-1; ans.length() < t; i+=m) {
            ans.append(sb.charAt(i));
        }
        
        return ans.toString().toUpperCase();
    }
}

File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.js
function solution(n, t, m, p) {
    var answer = '';
    let str = '';
    for (let i = 0; i <= t * t; i++) {
        str += i.toString(n);
    }
    for (let i = 0; i < str.length; i++) {
        if ((i+1) % m === p % m) {
            answer += str[i];
        }
    }
    answer = answer.slice(0,t).toUpperCase();
    return answer;
}